modules/mobile/config/routes.rb

```diff
@@ -49,7 +49,27 @@
     get '/letters/beneficiary', to: 'letters#beneficiary'
     post '/letters/:type/download', to: 'letters#download'
     get '/maintenance_windows', to: 'maintenance_windows#index'
+    get '/messaging/health/folders', to: 'folders#index'
+    post '/messaging/health/folders', to: 'folders#create'
+    get '/messaging/health/folders/:folder_id/messages', to: 'messages#index'
+    get '/messaging/health/folders/:folder_id/threads', to: 'threads#index'
+    get '/messaging/health/folders/:id', to: 'folders#show', as: 'folder'
+    delete '/messaging/health/folders/:id', to: 'folders#destroy'
+    post '/messaging/health/messages', to: 'messages#create'
+    get '/messaging/health/messages/categories', to: 'messages#categories'
     get '/messaging/health/messages/signature', to: 'messages#signature'
```

Comment:
due to how we structured these urls, this cannot be put alphabetically because if it is after the  `/messaging/health/messages/:id` then all signature endpoints will be routed to `/messaging/health/messages/:id` thinking that `signature` is the `:id`. Issues like this would be an argument for why we may want to follow the rails ways of doing things with resources rather then hardcoding things.

Reply:
This seems like a gotcha that could end up being annoying. That's one of the downsides to not leveraging the Rails Way™ of doing things. For instance, the messaging routes could be written like this:

```
namespace :messaging do
   namespace :health do
       resources :messages, only: :show do
           get :signature, on: :collection
           patch :move, on: :member
           post :reply, on: :member
           get :thread, on: :member
       end
   end
end
```
or even this
```
namespace :messaging do
    namespace :health do
        resources :messages, only: :show do
            collection do
                get :signature
            end 

            member do
                patch :move
                post :reply
                get :thread
            end
        end
    end
end
```
and Rails will know the correct ordering these routes need to be defined in so that collisions like @aherzberg mentioned don't happen. I definitely understand that parsing through routes written that way requires knowledge of routing DSL. On the other hand, we have our API doc site for a reason. It also looks like most (I didn't check them all) route files follow standard rails DSL.

Reply:
I'm not sure if I'm reading this conversation correctly, but it seems to have an undertone of "low confidence that this is the correct thing to do". It often does turn out that bucking convention causes unexpected issues. TBH, I've never worked anywhere that's written routes out individually like we're doing. If you feel confident everything is correct, I'm good with these changes. If you don't feel confident, I think it's fair to push back. If anyone struggles reading the routes file, they can just output rails routes.

Reply:
I have no issues with adhering to the rails way. The main point of this ticket was just to make all the routes use the same style of routing. If the individual listings is annoying or painful then let's do it whatever way makes the most sense.

Reply:
I'm confident these changes won't break any functionality and for consistency sake, I'll go ahead and merge this in. But for any new repos we create for internal tools, I think it'll be worth doing it the rails way.

Reply:
Hmmm after going through and trying to convert the routes, there are some really ugly routes we'd have to use the rails DSL for. Some of our route names drastically deviate from conventional rails routing so that makes it difficult to do this the proper way. Other modules have more closely followed conventional route names so that's partly why they are able to stick with the rails routing DSL. I'm good with merging as is and tests should check regressions if we ever touch the route file and accidentally mess up the order